instagram/src/components/CommentSection.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Comment from "./Comment";
+import CommentInput from "./CommentInput";
+
+class CommentSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      comments: props.comments,
+      comment: ""
+    };
+  }
+
+  commentHandler = e => {
+    this.setState({ comment: e.target.value });
+  };
+
+  commentSubmit = e => {
+    e.preventDefault();
+    // build out our comment obj
+    const newComment = { text: this.state.comment, username: "Leeroy Jenkins" };
+    // clone our comments array
+    const comments = this.state.comments.slice();
+    // push obj into new clone
+    comments.push(newComment);
+    // set new clone as state... + reset our comment str
+    this.setState({ comment: "", comments });
```

Comment:
Very nice. Experiment with seeing if you can refactor this to use `functional setState` for this, just so that you can learn how to become comfortable with it. If you need help, let me know.